TRRandomizerCore/Processors/TR3/TR3SequenceProcessor.cs

```diff
@@ -205,6 +206,29 @@ private void ImportUPV(TR3CombinedLevel level)
             }
         }
 
+        private void AddColdLevelMedis(TR3CombinedLevel level)
+        {
+            if (!level.Data.Entities.Any(e => e.TypeID == (short)TR3Entities.UPV))
+            {
+                return;
```

Comment:
https://github.com/LostArtefacts/TR-Rando/blob/4573c373db59760d4fa782165a7f0636fdaca842/TRRandomizerCore/Processors/TR3/TR3SequenceProcessor.cs#L92-L98

This is for levels that have freezing water but aren't antarctica or rx tech mines. These levels get both UPV and extra meds. At least that's how I read it; I'm not sure but it sounds like it should be the other way around.

Reply:
We don't add UPVs in all cases - Caves of Kaliya, Puna etc. So we don't want to add medis in these cases as they're intended mainly for getting off the UPV to pull a lever for instance. We could by all means, but it'd be free medis for no reason.

We don't do anything special in Antarctica or RX-Tech as they're cold OG, so we don't need to fiddle with the design really.

I suppose we could do a water room check instead of whether or not it's been decided previously to add a UPV, but this way it leaves it more loose - if we decide to add to other levels, it's just a case of defining vehicle locations [here](https://github.com/LostArtefacts/TR-Rando/blob/master/TRRandomizerCore/Resources/TR3/Locations/upv_locations.json).